elastalert/elastalert.py

```diff
@@ -1182,6 +1194,8 @@ def send_notification_email(self, text='', exception=None, rule=None, subject=No
         email = MIMEText(email_body)
         email['Subject'] = subject if subject else 'ElastAlert notification'
         email['To'] = ', '.join(self.notify_email)
+        if rule and rule['notify_email']:
```

Comment:
Can you check here that ``rule['notify_email']`` is not the same as ``self.notify_email``?